inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp

```diff
@@ -799,6 +799,10 @@ void MKLDNNGraph::Infer(int batch) {
         }
 
         ENABLE_DUMP(do_after(DUMP_DIR, graphNodes[i]));
+
+        if (IsCancellationRequested()) {
+            THROW_IE_EXCEPTION << InferenceEngine::details::as_status << InferenceEngine::INFER_CANCELLED;
```

Comment:
Why do you throw an exception?
This is conventional situation of **canceling the request in the middle of execution**.

Reply:
To return INFER_CANCELLED to callback function. Don't see  another ways. Infer has no return type.

Reply:
If i don't miss  something this is the only  way to return StatusCode

https://github.com/openvinotoolkit/openvino/blob/d617f1c2300b1cc21a35628888414f92932e1757/inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_async_request_thread_safe_default.hpp#L331